modules/tinymce/src/plugins/lists/main/ts/api/Commands.ts

```diff
@@ -20,7 +20,7 @@ const registerDialog = (editor: Editor): void => {
 };
 
 const register = (editor: Editor): void => {
-  editor.on('BeforeExecCommand', (e) => {
+  editor.on('ExecCommand', (e) => {
```

Comment:
This isn't the correct way to handle this case, as we want to override the default behavior and this would cause it to run after the default behavior instead. So what we should be doing to solve this is to add a check to see if it's already been prevented instead. As such this probably needs to be something like this:
```
editor.on('BeforeExecCommand', (e) => {
  if (!e.isDefaultPrevented()) {
    const cmd = e.command.toLowerCase();

    if (cmd === 'indent') {
      indentListSelection(editor);
    } else if (cmd === 'outdent') {
      outdentListSelection(editor);
    }
  }
});
```

Reply:
Don't plugins register their handlers before the core? So swapping to `ExecCommand` will still override the core indent/outdent behaviour? It seems strange to me that lists had to use `Before` to achieve the override.

Reply:
No, that's not how events, nor the commands work. We have the `BeforeExecCommand` event which fires before the core or native commands run and then the `ExecCommand` event which fires after that. So in this case the flow of events is:
- `BeforeExecCommand` event is fired
  - List indent/outdent behaviour override
- Core indent/outdent command runs
- `ExecCommand` event is fired

I'm really not sure how you expect the `ExecCommand` event to run before depending on the plugin registration order to be honest. I'm wondering if you've got mixed up thinking Core does the processing in an event, rather than in the command? Worth noting that we also can't just override the indent/outdent commands, as it only needs to override in specific cases.